message-index/messages/GHC-99339/singlemodule/index.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Ambiguous record update (single module)
+---
+
+Both records `R1` and `R2` share the same field `x`. For this reason, the compiler cannot infer a type for the record update `r { x = 1 }`.
+If both records are declared in the same module, then the forward compatible way to solve this error is to rename one of the fields.
```

Comment:
...or use a case expression, perhaps with `NamedFieldPuns`/`RecordWildCards` for brevity.

Reply:
I added the alternative in  f8b47d5, but the alternative without using an additional language extension.